### PR TITLE
Fix warnings and small fixes

### PR DESCRIPTION
**Pull request by ausyskin**

### CODE REVIEW

File: src/server/shttpd/io_emb.c [hunk 1]
```diff
@@ -306,7 +306,7 @@ shttpd_wakeup(const void *priv)
 	conn->flags &= ~SHTTPD_SUSPEND;
 #endif
 	(void) memcpy(buf, &cmd, sizeof(cmd));
-	(void) memcpy(buf + sizeof(cmd), conn, sizeof(conn));
+	(void) memcpy(buf + sizeof(cmd), &conn, sizeof(conn));
```
Comment (kkaempf) [inline src/server/shttpd/io_emb.c]: Hmm, this looks wrong. conn is declared as `const struct conn       *`, so it is already a pointer and the original `memcpy` call looks right to me.

Comment (kkaempf) [reply src/server/shttpd/io_emb.c]: The `sizeof` call needs to be changed instead and read `sizeof(struct conn)`

Comment (ausyskin) [reply src/server/shttpd/io_emb.c]: Yep, the fix look wrong now, but I think I know why static analyzer jumped: sizeof(conn) is sizeof of pointer, not the structure, sod sizeof(conn) -> sizeof(struct conn) should help here

File: src/lib/wsman-filter.c [hunk 2]
```diff
@@ -161,7 +161,7 @@ filter_t * filter_create_selector(hash_t *selectors)
 static int
 _filter_add_selector(filter_t *filter, const char* key, const char *value, const epr_t *epr)
 {
-	int i;
+	unsigned int i;
```
Comment (kkaempf) [inline src/lib/wsman-filter.c]: while you're at it, please move the declaration of `i` into the `if (filter_cpy->selectorset.selectors) {` block. Just like it's done for the `if(filter->resultProp) {` block in the same function.

File: src/server/shttpd/io_emb.c (hunk 1 above)
Comment (ausyskin) [reply src/server/shttpd/io_emb.c]: Or better rename conn pointer to something else to avoid name collision.

File: src/lib/wsman-filter.c [hunk 3]
```diff
@@ -357,7 +357,8 @@ filter_t * filter_deserialize(WsXmlNodeH node, const char *ns)
 {
 	char *dialect = NULL;
 	int properNum = 0;
-	int i = 0;
+	unsigned int i = 0;
```
Comment (kkaempf) [inline src/lib/wsman-filter.c]: the declaration (and initialization) of `i` should be moved to the block of actual use (the `if(strcmp(filter->dialect , WSM_SELECTOR_FILTER_DIALECT) == 0) {` one)

File: src/lib/wsman-filter.c [hunk 4]
```diff
@@ -357,7 +357,8 @@ filter_t * filter_deserialize(WsXmlNodeH node, const char *ns)
...
+	int j = 0;
```
Comment (kkaempf) [inline src/lib/wsman-filter.c]: Then this line can go to the `if(strcmp(filter->dialect , WSM_ASSOCIATION_FILTER_DIALECT) == 0) {` block

File: src/server/shttpd/io_emb.c [hunk 5]
```diff
@@ -306,7 +306,7 @@ shttpd_wakeup(const void *priv)
 	conn->flags &= ~SHTTPD_SUSPEND;
 #endif
 	(void) memcpy(buf, &cmd, sizeof(cmd));
-	(void) memcpy(buf + sizeof(cmd), conn, sizeof(conn));
+	(void) memcpy(buf + sizeof(cmd), conn, sizeof(struct conn));
```
Comment (kkaempf) [inline src/server/shttpd/io_emb.c]: Argh, I should've looked closer before. This is some really ugly piece of code in `shttpd`. Sorry for the confusion.

Problem: The `buf` memcpy is using here is sized to `sizeof(int) + sizeof(void *)`. So it only has space for a _pointer_ (to `struct conn`) but not for the whole struct.
This means that your first fix (adding `&` to the `conn` parameter of `memcpy`) was actually right.

Comment (kkaempf) [reply src/server/shttpd/io_emb.c]: Biggest question here: How to test this ? :-/

File: src/server/shttpd/io_emb.c (hunk 1 above)
Comment (kkaempf) [reply src/server/shttpd/io_emb.c]: I hate `shttpd` :stuck_out_tongue_winking_eye:

**Comment by kkaempf** [comment]
Thanks !!

**Pull request merged**